back/spec/acceptance/ideas_create_spec.rb

```diff
@@ -561,13 +565,84 @@ def public_input_params(spec)
+            assert_status 201
+            idea_from_db = Idea.find(response_data[:id])
+            expect(idea_from_db.author_id).to be_nil
+            expect(idea_from_db.custom_field_values.to_h).to eq({
+              extra_field_name => 'test value'
+            })
+          end
+        end
+
+        context 'response has already been submitted' do
+          let!(:author_hash) do
+            # No consent hash based on ip and user agent
+            user_agent = 'User-Agent: Mozilla/5.0'
+            ip = '1.2.3.4'
+            "n_#{Idea.create_author_hash(ip + user_agent, project.id, true)}"
+          end
+          let!(:response) { create(:native_survey_response, project: project, creation_phase: phase, author: nil, author_hash: author_hash) }
+
+          context 'no cookie present - using headers' do
+            example 'does not allow posting if submitted within 3 months' do
+              response.update!(published_at: 2.months.ago)
+              header 'User-Agent', 'User-Agent: Mozilla/5.0'
+              header 'X-Forwarded-For', '1.2.3.4'
+              do_request
+              assert_status 401
+              expect(json_response_body.dig(:errors, :base, 0, :error)).to eq 'posting_limited_max_reached'
+            end
+
+            example 'allows posting again if submitted after 3 months' do
+              response.update!(published_at: 4.months.ago)
+              header 'User-Agent', 'User-Agent: Mozilla/5.0'
+              header 'X-Forwarded-For', '1.2.3.4'
+              do_request
+              assert_status 201
+            end
+          end
+
+          context 'cookie is present' do
+            let!(:author_hash) { 'LOGGED_OUT_HASH' }
+
+            example 'does not allow posting if user submitted survey without cookie consent (empty cookie present)' do
+              header('Cookie', "#{phase.id}={}")
+              do_request
+              assert_status 401
+              expect(json_response_body.dig(:errors, :base, 0, :error)).to eq 'posting_limited_max_reached'
+            end
+
+            example 'does not allow posting if submitted within 3 months' do
+              response.update!(published_at: 2.months.ago)
+              header('Cookie', "#{phase.id}={\"lo\": \"LOGGED_OUT_HASH\"};cl2_consent={\"analytics\": true}")
+              do_request
+              assert_status 401
+              expect(json_response_body.dig(:errors, :base, 0, :error)).to eq 'posting_limited_max_reached'
+            end
+
+            example 'allows posting again if submitted after 3 months' do
+              response.update!(published_at: 4.months.ago)
+              header('Cookie', "#{phase.id}={\"lo\": \"LOGGED_OUT_HASH\"};cl2_consent={\"analytics\": true}")
+              do_request
+              assert_status 201
+              # Saves a new idea with the same author hash from the cookie
+              expect(Idea.all.pluck(:author_hash)).to eq %w[LOGGED_OUT_HASH LOGGED_OUT_HASH]
```

Comment:
I think this could become a flaky spec (no explicit sorting of the ideas scope), I think we need to either apply an ordering or use `match_array`.